Library/Homebrew/resource.rb

```diff
@@ -261,6 +212,14 @@ def patch(strip = :p1, src = nil, &block)
     patches << p
   end
 
+  def using
+    @url&.using
```

Comment:
Same here: when would we want a resource with a `nil` URL? Could we lean on the type system for that?